### PR TITLE
Added support to retrieve system properties from upstream python-dracclient

### PR DESCRIPTION
**Pull request by uday344**
@cdearborn @rpioso 

1. Removed existing code from discover_nodes.
2. Added code to support retrieval of system properties from upstream python-dracclient.

### CODE REVIEW

**Comment by digambar15** [comment]
We need a topic branch to create a PR. Creating PR against Master branch will be risk here. 
Like we merged all the PR against UEFI topic branch and ran all the test cases and once we got successful test run then merged UEFI experimental branch against Master.

**Comment by gaelrehault** [comment]
@uday344 can you create the PR  against the experimental_upstream_ironic branch please

File: src/pilot/assign_role.py [hunk 1]
```diff
@@ -1553,7 +1557,8 @@ def main():
             args.role_index,
             args.os_volume_size_gb,
             ironic_client,
-            drac_client)
+            args.node_definition,
```
Comment (cdearborn) [inline src/pilot/assign_role.py]: Can you explain the reasoning behind this message signature change?  It looks like passing in drac_client should work.

File: src/pilot/assign_role.py [hunk 2]
```diff
@@ -861,9 +862,12 @@ def place_node_in_available_state(ironic_client, node_uuid):
 
 
 def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
-                ironic_client, drac_client):
+                ironic_client, node_definition, node):
     flavor = ROLES[role_index.role]
-
+    drac_ip, drac_user, drac_password = \
```
Comment (cdearborn) [inline src/pilot/assign_role.py]: I don't understand why drac_client is being rebuilt here.  Can't it just be passed in by the caller?

File: src/pilot/config_idrac.py [hunk 3]
```diff
@@ -68,10 +69,16 @@ def parse_arguments():
     return parser.parse_args()
 
 
-def get_pxe_nic_fqdd(fqdd, model_properties, drac_client):
+def get_pxe_nic_fqdd(fqdd, model_properties, ip_service_tag, node_definition):
```
Comment (cdearborn) [inline src/pilot/config_idrac.py]: What is the reasoning behind changing the method signature here?  It seems much cleaner to leave it as-is and pass drac_client.

File: src/pilot/config_idrac.py [hunk 4]
```diff
@@ -405,7 +412,8 @@ def config_idrac(instack_lock,
     pxe_nic_fqdd = get_pxe_nic_fqdd(
         pxe_nic,
         model_properties,
-        drac_client)
+        ip_service_tag,
```
Comment (cdearborn) [inline src/pilot/config_idrac.py]: I don't understand why it's necessary to change the method signature here.

**Comment by uday344** [comment]
Added changes and created new PR on experimental_upstream_ironic https://github.com/dsp-jetpack/JetPack/pull/202
Hence closing this PR

File: src/pilot/assign_role.py (hunk 2 above)
Comment (uday344) [reply src/pilot/assign_role.py]: Modified this code 
Please refer https://github.com/dsp-jetpack/JetPack/pull/202

File: src/pilot/assign_role.py (hunk 1 above)
Comment (uday344) [reply src/pilot/assign_role.py]: Modified this code 
Please refer https://github.com/dsp-jetpack/JetPack/pull/202

File: src/pilot/config_idrac.py (hunk 3 above)
Comment (uday344) [reply src/pilot/config_idrac.py]: Upstream drac client does not contain all methods which were used in config_idrac.py.
like 
https://github.com/dsp-jetpack/JetPack/blob/81fff38e72ec60e156934cf1d0363a91e49662ef/src/pilot/config_idrac.py#L173
and 
https://github.com/dsp-jetpack/JetPack/blob/81fff38e72ec60e156934cf1d0363a91e49662ef/src/pilot/config_idrac.py#L176

So the code needs to have 2 drac clients upstream and discover nodes local drac.
For now I have added one upstream, in future when we add all methods in upstream drac then we can remove this code.

Please continue discussion on 
https://github.com/dsp-jetpack/JetPack/pull/202

File: src/pilot/config_idrac.py (hunk 4 above)
Comment (uday344) [reply src/pilot/config_idrac.py]: Answered this on line 72

**Pull request closed without merging**